src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -0,0 +1,126 @@
+package seedu.address.model.appointment;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.model.datetime.DateTime;
+import seedu.address.model.datetime.RecurringDateTime;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Person;
+
+/**
+ * Represents an Appointment in the application
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Appointment {
+
+    // Identity fields
+    private final DateTime startDateTime;
+    private final DateTime endDateTime;
+    private final RecurringDateTime frequency;
+
+    // Data fields
+    private final Person patient;
+    private final String description;
+    // private final List<Medicine> medicinePackingList;
+
+    public Appointment(DateTime startDateTime, DateTime endDateTime, RecurringDateTime frequency,
+                       Person patient, String description) {
+        requireAllNonNull(startDateTime, endDateTime, frequency, patient, description);
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
+        this.frequency = frequency;
+        this.patient = patient;
+        this.description = description;
+    }
+
+    public DateTime getStartDateTime() {
+        return startDateTime;
+    }
+
+    public DateTime getEndDateTime() {
+        return endDateTime;
+    }
+
+    public RecurringDateTime getFrequency() {
+        return frequency;
+    }
+
+    public Person getPatient() {
+        return patient;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public Address getLocation() {
+        return patient.getAddress();
+    }
+
+    /**
+     * Returns true if both appointments of the same date and time have at least one other identity field that is the
+     * same.
+     * This defines a weaker notion of equality between two appointments.
+     */
+    public boolean isSameAppointment(Appointment otherAppointment) {
+        if (otherAppointment == this) {
+            return true;
+        }
+
+        return otherAppointment != null
+                && otherAppointment.getStartDateTime().equals(getStartDateTime())
+                && (otherAppointment.getEndDateTime().equals(getEndDateTime())
+                    || otherAppointment.getFrequency().equals(getFrequency()));
+    }
+
+    /**
+     * Returns true if both appointments have the same identity and data fields.
+     * This defines a stronger notion of equality between two appointments.
+     */
+    @Override
+    public boolean equals(Object other) {
```

Comment:
The other isSameAppointment function essentially checks if two appointments occur at the same time.
Is having two seperate functions entirely necessary?
If it is essential to your future implementation, perhaps changing the other function to isSameTimeslot would provide greater clarity.

Reply:
Agreed, I am not sure at the moment if I would need it later on, but the Person class had similar distinctions between a weaker and stricter equality so decided to adopt the same for now. Can remove later if it is not used.